src/Agent.Plugins/PipelineArtifact/ArtifactProviderFactory.cs

```diff
@@ -13,11 +13,13 @@ internal class ArtifactProviderFactory
     {
         private readonly FileContainerProvider fileContainerProvider;
         private readonly PipelineArtifactProvider pipelineArtifactProvider;
+        private readonly FileShareProvider fileShareProvider;
```

Comment:
not sure why this conversation was resolved but where is the code for FileShareProvider?

Reply:
?? FileShareProvider should already be added

Reply:
That's right, my mistake I saw the logic now and it's fine the way it is.